src/pybroker/ext/data.py

```diff
@@ -91,3 +92,101 @@ def _fetch_data(
             ]
         ]
         return result
+
+
+class YQDataSource(DataSource):
+    r"""Retrieves data from `Yahoo Finance <https://finance.yahoo.com/>`_\ .
+
+    Attributes:
+        ADJ_CLOSE: Column name of adjusted close prices.
+    """
+
+    ADJ_CLOSE: Final = "adj_close"
+    __TIMEFRAME: Final = "1d"
+    _tf_to_period = {
```

Comment:
Are these the only time frames supported by YQ? Or does it support others like 20m, 45m etc?

Note, there is a common.py#parse_timeframe function that will return a tuple of value and units. With that, we could map w -> wk.

Reply:
Actually, according to its [doc](https://yahooquery.dpguthrie.com/guide/ticker/historical/), it supports many other time frames/interval, but it's limited to ['1m', '2m', '5m', '15m', '30m', '60m', '90m', '1h', '1d', '5d', '1wk', '1mo', '3mo'].


`Note, there is a common.py#parse_timeframe function that will return a tuple of value and units. With that, we could map w -> wk.`

Yes, I noticed that function, but it generates result as "1d" -> "1day", which YQ won't accept as parameter, and I don't want to make other changes to your original code, so I created a dict "_tf_to_period" and did mapping at line 168.